lib/propolis/src/hw/nvme/admin.rs

```diff
@@ -282,4 +293,56 @@ impl NvmeCtrl {
             }
         }
     }
+
+    /// Write result data from an admin command into host memory
+    ///
+    /// Returns `Some(())` if successful, else None
+    fn write_admin_result<T: Copy>(
```

Comment:
It would be neat to enforce the packed-ness of T here, but the most promising looking crate I found to do so (`repr-trait`) doesn't quite look to me like it would drop into this code neatly (reading through its derives I'm not convinced it will handle the `repr(C, packed(1))` repr that NVMe's structs use). Oh well--maybe another time.

Reply:
Alternatively, a `write_from_ptr()` interface would keep us clear of the potential UB from the slice creation.  This should work for now.

Reply:
Added to the comment regarding the `repr()` expectations, so it's at least a bit more visible in the mean time.